File: 23xxx/23757. 아이들과 선물 상자/아이들과 선물 상자.py
from queue import PriorityQueue

N, M = map(int, input().split())
pq = PriorityQueue()

present = list(map(int, input().split()))
for e in present:
    pq.put(-e)

w = list(map(int, input().split()))
for want in w:
    top = -pq.get()
    if top<want:
        print(0)
        exit()
    if top-want>0:
        pq.put(-(top-want))
print(1)

File: 23xxx/23757. 아이들과 선물 상자/아이들과 선물 상자.cc
#include<bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0); cin.tie(0); cout.tie(0);
    int N, M; cin >> N >> M;

    priority_queue<int, vector<int>, less<int>> pq;
    while(N-->0) {
        int present; cin >> present;
        pq.push(present);
    }

    while(M-->0) {
        int want; cin >> want;
        if(!pq.empty() && pq.top()>=want) {
            int top = pq.top(); pq.pop();
            if(top-want>0) pq.push(top-want);
        } else {
            cout << 0;
            return 0;
        }
    }
    cout << 1;
}